### PR TITLE
Search Form Autocomplete

### PR DESCRIPTION
**Pull request by zlister**
This is a PR that completes the autocomplete feature for the search form.

### CODE REVIEW

**Review by dmpotter44 — CHANGES_REQUESTED** [review]
The most immediate thing are all the lint failures. They should be resolved before proceeding. Some of them can be automatically fixed but some of them are going to require a bit more effort to properly define types.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    '',
+    MultipleMyelomaMedicationCodes,
+    '',
+    '',
+    breastCancerStages,
+    ''
+  ),
+};
+
+export const cancerTypeDetails: cancerTypeObj = tmp;
```
Comment (dmpotter44) [inline src/components/SearchForm/SearchFormOptions.tsx:112]: I'm unclear why the `const tmp` and why not just `export const cancerTypeDetails: cancerTypeObj`.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+export const breastCancerStages = [0, 1, 2.1, 2.2, 3.1, 3.2, 3.3, 4];
+//export const breastCancer_cancerTypeOptions=buildOptionListfromJSON(breastcancerCodes);
+export const breastCancerMedications = '';
+export const breastCancerSurgeryCodes = '';
+export const breastCancer_proceduresOptions = '';
+export const breastCancerBiomarkers = '';
+export const breastCancer_biomarkersOptions = '';
+export const breastCancerRadiationCodes = '';
+export const breastCancerSubTypes = '';
+export const breastCancerCodes = '';
```
Comment (dmpotter44) [inline src/components/SearchForm/SearchFormOptions.tsx:38]: I would argue constants like this should be in `CAPITALS_WITH_UNDERSCORES`.

**Review by dmpotter44 — COMMENTED** [review]
I expect that there's probably more to go over but we should address the lint and style changes first.

**Review by jtquach1 — CHANGES_REQUESTED** [review]
Additionally, please remove `node_modules` completely and then do `yarn install` so there aren't any potential conflicts with existing installed dependencies and so the Prettier pre-commit hooks are properly installed. The linting issues related to imports and quotes should be automatically resolved by the Prettier pre-commit hook after pushing a commit.

File: src/components/SearchForm/FormFields.tsx [hunk 3]
```diff
@@ -81,22 +70,39 @@ export const AgeTextField = ({
 
 export const CancerTypeAutocomplete = ({
   field,
+  retrieveCancer,
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'cancerType'>;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  retrieveCancer: Function;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: Modified to "(value: string) => void"

File: src/components/SearchForm/FormFields.tsx [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
     {...field}
+    disabled={options === undefined || options === null || options?.length === 0 ? true : false}
     data-testid={label}
     freeSolo
     multiple
-    onChange={(_event, value) => field.onChange(value)}
+    onChange={(event, newValue) => {
+      field.onChange(newValue);
+    }}
+    getOptionLabel={option => option?.display}
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: Modified to "getOptionLabel={option => option}" as suggested.

File: src/components/SearchForm/SearchForm.tsx [hunk 5]
```diff
@@ -1,11 +1,14 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: resolved with installation of prettier in vscode.

File: src/components/SearchForm/FormFields.tsx [hunk 6]
```diff
@@ -81,22 +70,39 @@ export const AgeTextField = ({
...
+      enabled: typeof window !== 'undefined',
+   });*/
+  // console.log(cancerTypeOptions)
+  //const { data, isLoading } = cancerTypeOptions;
+  // const [initialValue] = useState(field.value);
+  // const options = useMemo(() => [initialValue, ...(data || [])].filter(Boolean), [initialValue, cancerTypeOptions]);
+  const newOptions = [];
+  for (const [key, value] of Object.entries(cancerTypeOptions[0])) {
+    for (const [key2, cancerTypes] of Object.entries(value['cancerCodes'])) {
+      cancerTypes.forEach(element => {
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: resolved.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+function buildCancerCodeJSON(
+  category,
+  cancerCodes,
+  subTypeCodes,
+  medicationCodes,
+  radiationCodes,
+  surgeryCodes,
+  stages,
+  biomarkerCodes
+) {
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: added type definitions

File: src/utils/fhirFilter.ts [hunk 8]
```diff
@@ -143,3 +144,63 @@ export function convertStringtoResource({
...
+  codedValue: NamedSNOMEDCode;
+  id: string;
+  profile_value: string;
+  codingSystem: string;
+  codingSystemCode: string;
+}): void {
+  // Create the Condition - done separate from the function call to ensure proper TypeScript checking
+
+  const tmpCode: string | number = codedValue.code.toString();
+  const tmpDisplay = codedValue.display;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/utils/fhirFilter.ts:170]: Made suggested modification to code.

File: types/search-types.ts [hunk 9]
```diff
@@ -1,3 +1,5 @@
+import { NamedSNOMEDCode } from "@/utils/fhirConversionUtils";
```
[the remark this answers is not in the document]
Comment (napolr1) [reply types/search-types.ts]: fixed with installation of prettier.

File: src/components/SearchForm/FormFields.tsx [hunk 10]
```diff
@@ -131,14 +140,15 @@ export const CancerSubtypeAutocomplete = ({
 
 export const CancerStageAutocomplete = ({
   field,
+  canceStages,
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'stage'>;
+  canceStages: any;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: Typo Corrected.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+//export const breastCancer_cancerTypeOptions=buildOptionListfromJSON(breastcancerCodes);
+export const breastCancerMedications = '';
+export const breastCancerSurgeryCodes = '';
+export const breastCancer_proceduresOptions = '';
+export const breastCancerBiomarkers = '';
+export const breastCancer_biomarkersOptions = '';
+export const breastCancerRadiationCodes = '';
+export const breastCancerSubTypes = '';
+export const breastCancerCodes = '';
+export type cancerTypeDetails = {
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: modified as suggested.

File: src/components/SearchForm/SearchFormOptions.tsx (hunk 1 above)
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: removed tmp variable as suggested.

File: src/components/SearchForm/SearchFormOptions.tsx (hunk 2 above)
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: Modified code to reflect suggestion of CAPITALS_WITH_UNDERSCORES

File: .prettierrc [hunk 12]
```diff
@@ -1,6 +1,11 @@
...
   "singleQuote": true,
+
   "printWidth": 120,
+
   "trailingComma": "es5",
-  "arrowParens": "avoid"
+
+  "arrowParens": "avoid",
+
+  "endOfLine": "lf"
```
Comment (dmpotter44) [inline .prettierrc:10]: I think this file has extra newlines in it there were added by a CRLF conversion at some point.

File: src/components/SearchForm/FormFields.tsx [hunk 13]
```diff
@@ -176,37 +184,42 @@ export const MetastasisAutocomplete = ({
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'metastasis'>;
 }): ReactElement => (
+  // eslint-disable-next-line prettier/prettier
```
Comment (dmpotter44) [inline src/components/SearchForm/FormFields.tsx:187]: Is there a reason Prettier is disabled for this line? If there is, it should be mentioned with a comment.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+    'mm',
+    MULTIPLE_MYELOMA_CANCERTYPE_CODES.entry,
+    null,
+    MULTIPLE_MYELOMA_MEDICATION_CODES.entry,
+    null,
+    null,
+    BREAST_CANCER_STAGES.entry,
+    null
+  ),
+};
```
Comment (dmpotter44) [inline src/components/SearchForm/SearchFormOptions.tsx:100]: Either the nulled values should have comments explaining why they aren't needed, or the function should be refactored to take an options object that only contains the required keys.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+    cancerSubtype: subTypeCodes,
+    biomarkers: biomarkerCodes,
+    medications: medicationCodes,
+    radiationCodes: radiationCodes,
+    surgeryCodes: surgeryCodes,
+    stages: stages,
+  };
+
+  return details;
+}
```
Comment (dmpotter44) [inline src/components/SearchForm/SearchFormOptions.tsx:132]: I don't understand the point of this function. Why not just use the object it declares directly?

File: src/components/Sidebar/Sidebar.tsx [hunk 16]
```diff
@@ -56,16 +56,17 @@ const Sidebar = ({ patient, disabled, savedStudies, filterOptions }: SidebarProp
...
     ecogScore: (query.ecogScore as string) || null,
     karnofskyScore: (query.karnofskyScore as string) || null,
-    biomarkers: ensureArray(query.biomarkers),
-    radiation: ensureArray(query.radiation),
-    surgery: ensureArray(query.surgery),
-    medications: ensureArray(query.medications),
+    biomarkers: parseCodedValueArray(query.biomarkers as string),
+    radiation: parseCodedValueArray(query.radiation as string),
+    surgery: parseCodedValueArray(query.surgery as string),
+    medications: parseCodedValueArray(query.medications as string),
```
Comment (dmpotter44) [inline src/components/Sidebar/Sidebar.tsx:69]: I'm not sure why `as string` is required but that's an immediately code smell. The functions should be updated to use whatever the actual type is, and if it's a string, then the type of the object should be fixed.

File: src/pages/api/clinical-trial-search.ts [hunk 17]
```diff
@@ -147,78 +155,81 @@ function buildBundle(searchParams: SearchParameters): Bundle {
...
-    const codingSystemCode = '21907-1';
-    convertStringtoResource({
-      bundle: patientBundle,
-      valueString: metastasisParm,
-      id,
-      profile_value: profileValue,
-      codingSystem,
-      codingSystemCode,
-    });
+    const codingSystem = '';
```
Comment (dmpotter44) [inline src/pages/api/clinical-trial-search.ts:173]: Why is this `const` an empty string?

**Review by dmpotter44 — CHANGES_REQUESTED** [review]
`src/queries/mockData.zipx` still needs to be deleted. There are still a bunch of comments that need to be resolved that haven't been yet.

File: src/utils/fhirFilter.ts [hunk 18]
```diff
@@ -143,3 +144,63 @@ export function convertStringtoResource({
...
+  codingSystemCode,
+}: {
+  bundle: Bundle;
+  codedValue: NamedSNOMEDCode;
+  id: string;
+  profile_value: string;
+  codingSystem: string;
+  codingSystemCode: string;
+}): void {
+  // Create the Condition - done separate from the function call to ensure proper TypeScript checking
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/utils/fhirFilter.ts:158]: Split function into 2 separate functions and re-configured it to return a resource.

File: .prettierrc (hunk 12 above)
Comment (napolr1) [reply .prettierrc:10]: Deleted src/queries/mockData.zipx

File: src/components/Sidebar/Sidebar.tsx (hunk 16 above)
Comment (dmpotter44) [reply src/components/Sidebar/Sidebar.tsx]: I've pushed a change that resolves this by allowing the parse methods to accept arrays.

File: src/components/SearchForm/FormFields.tsx [hunk 19]
```diff
@@ -181,32 +191,36 @@ export const MetastasisAutocomplete = ({
...
 
 export const SurgeryAutocomplete = ({
   field,
+  cancerSurgery,
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'surgery'>;
-}): ReactElement => (
-  <AutocompleteMulti field={field} label="surgery" options={['surgery-1', 'surgery-2', 'surgery-3']} />
-);
+  cancerSurgery: any;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: Assigned cancerSurgery type to  CodedValueType[]

File: src/utils/fhirFilter.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (napolr1) [reply src/utils/fhirFilter.ts:158]: Function re-written (split ) into two functions. One returns an observation resource, the other a medicationStatement resource

File: src/components/SearchForm/SearchFormOptions.tsx (hunk 15 above)
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx:132]: removed function and created the objects directly

File: src/components/SearchForm/SearchFormOptions.tsx (hunk 14 above)
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx:100]: replaced nulls with empty arrays.  Added comments where appropriate

File: src/pages/api/clinical-trial-search.ts (hunk 17 above)
Comment (napolr1) [reply src/pages/api/clinical-trial-search.ts:173]: replaced empty string with appropriate coding system codes.

File: .prettierrc (hunk 12 above)
Comment (dmpotter44) [reply .prettierrc:10]: That doesn't resolve *this* issue, where the `.prettierrc` file has extra new lines.

File: src/components/SearchForm/FormFields.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmpotter44) [reply src/components/SearchForm/FormFields.tsx:86]: Has this change not been checked in?

**Comment by napolr1** [comment]
I didn’t see your push.  Let me check for your update


From: Dan Potter ***@***.***>
Sent: Wednesday, September 21, 2022 2:53 PM
To: mcode/clinical-trial-matching-app ***@***.***>
Cc: Napoleon Russ ***@***.***>; Comment ***@***.***>
Subject: [EXT] Re: [mcode/clinical-trial-matching-app] Search Form Autocomplete (PR #42)


@dmpotter44 commented on this pull request.

________________________________

In src/components/Sidebar/Sidebar.tsx<https://github.com/mcode/clinical-trial-matching-app/pull/42#discussion_r976883010>:

> +    cancerType: ensureCodedValueType(query.cancerType as string),

+    cancerSubtype: ensureCodedValueType(query.cancerSubtype as string),

     metastasis: ensureArray(query.metastasis),

-    stage: (query.stage as string) || null,

+    //stage: (query.stage as string) || null,

+    stage: ensureCodedValueType(query.stage as string),

     ecogScore: (query.ecogScore as string) || null,

     karnofskyScore: (query.karnofskyScore as string) || null,

-    biomarkers: ensureArray(query.biomarkers),

-    radiation: ensureArray(query.radiation),

-    surgery: ensureArray(query.surgery),

-    medications: ensureArray(query.medications),

+    biomarkers: parseCodedValueArray(query.biomarkers as string),

+    radiation: parseCodedValueArray(query.radiation as string),

+    surgery: parseCodedValueArray(query.surgery as string),

+    medications: parseCodedValueArray(query.medications as string),

Why is as string back?

File: src/components/SearchForm/FormFields.tsx [hunk 20]
```diff
@@ -181,32 +191,36 @@ export const MetastasisAutocomplete = ({
 
 export const BiomarkersAutocomplete = ({
   field,
+  cancerBiomarkers,
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'biomarkers'>;
-}): ReactElement => (
-  <AutocompleteMulti field={field} label="biomarkers" options={['biomarker-1', 'biomarker-2', 'biomarker-3']} />
-);
+  cancerBiomarkers;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: This was resolved previously.  The NamedSNOMEDCode type has been renamed to codedValueType and asseincancerBiomarkers has been modified to use that type

File: src/components/SearchForm/FormFields.tsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: cancerSurgery has been redefined as CodedValueType[] type.

File: src/components/SearchForm/FormFields.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx:86]: It looks as though this change was lost.  I've re-implemented this and will upload shortly.

File: src/components/SearchForm/FormFields.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: The variable name change and type change were previously modified. to...

cancerStages: CodedValueType[];

File: src/components/SearchForm/FormFields.tsx [hunk 21]
```diff
@@ -181,32 +191,36 @@ export const MetastasisAutocomplete = ({
...
 
 export const MedicationsAutocomplete = ({
   field,
+  cancerMedication,
 }: {
   field: ControllerRenderProps<SearchFormValuesType, 'medications'>;
-}): ReactElement => (
-  <AutocompleteMulti field={field} label="medications" options={['medication-1', 'medication-2', 'medication-3']} />
-);
+  cancerMedication: any;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/FormFields.tsx]: This "watch" statement has been removed.

File: src/components/SearchForm/SearchForm.tsx [hunk 22]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+      ...formDataToSearchQuery(data),
+      sortingOption: 'matchLikelihood',
+      page: DEFAULT_PAGE,
+      pageSize: DEFAULT_PAGE_SIZE,
+    },
+  });
+*/
+  //**** Russ */
+
+  const cancerType = watch('cancerType');
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: This "watch" statement has been removed.

File: src/components/SearchForm/SearchForm.tsx [hunk 23]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+    medication: '',
+    procedure: '',
+  });
+
+  const onChange = event => handleChange(cancerTypes);
+
+  const handleChange = e => {
+    const val = e.target.value;
+    setcancerCategory(cancerTypes['val'].entryType);
+    //console.log('cancerCategory=' + cancerCategory);
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: this line has been removed.

File: src/components/SearchForm/SearchForm.tsx [hunk 24]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+        setMedications([]);
+        setProcedures([]);
+        break;
+      case 'breast':
+        setCancerTypes(cancerTypeDetails.breast.cancerCodes);
+        setCancerSubTypes(cancerTypeDetails.brain.cancerSubtype);
+        //setBiomarkers(breastCancer_biomarkersOptions);
+        setStages(cancerTypeDetails.brain.stages);
+        //setMedications(breastCancer_medicationsOptions);
+        //setProcedures(breastCancer_proceduresOptions);
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: This code is no longer commented out.

File: src/components/SearchForm/SearchForm.tsx [hunk 25]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+        setCancerSubTypes(cancerTypeDetails.mm.cancerSubtype);
+        setStages(cancerTypeDetails.breast.stages);
+        setBiomarkers([]);
+        setStages([]);
+        setMedications([]);
+        setProcedures([]);
+    }
+  };
+
+  const retrieveCancer = cancer => {
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: Resolved - 

const retrieveCancer = (cancer: CodedValueType) => {

File: src/components/SearchForm/SearchForm.tsx [hunk 26]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+    if (cancer !== null) {
+      //console.log(cancerTypeOptions[0][cancer.entryType]);
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: All uneccesary console.log calls that I've added have been removed.

File: src/components/SearchForm/SearchForm.tsx [hunk 27]
```diff
@@ -38,22 +41,166 @@ export const formDataToSearchQuery = (data: SearchFormValuesType): SearchParamet
...
+      if (cancerTypeOptions[0][cancer.entryType] !== undefined) {
+        if (
+          cancerTypeOptions[0][cancer.entryType].cancerSubtype !== null ||
+          cancerTypeOptions[0][cancer.entryType].cancerSubtype?.length > 0
+        ) {
+          setCancerSubTypes(cancerTypeOptions[0][cancer.entryType].cancerSubtype?.entry);
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchForm.tsx]: I don't see a difference between the current syntax and suggested syntax.

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 28]
```diff
@@ -0,0 +1,145 @@
...
+import lungCancerSubTypeCodes from 'src/queries/mockData/lungCancerSubTypeCodes.json';
+import lungCancerTypeCodes from 'src/queries/mockData/lungCancerTypeCodes.json';
+import lungMedicationCodes from 'src/queries/mockData/lungMedicationCodes.json';
+import lungRadiationCodes from 'src/queries/mockData/lungRadiationCodes.json';
+import lungSurgeryCodes from 'src/queries/mockData/lungSurgeryCodes.json';
+import MultipleMyelomaCancerTypeCodes from 'src/queries/mockData/MultipleMyelomaCancerTypeCodes.json';
+import MultipleMyelomaMedicationCodes from 'src/queries/mockData/MultipleMyelomaMedicationCodes.json';
+import prostateCancerTypeCodes from 'src/queries/mockData/prostateCancerTypeCodes.json';
+
+//import type { NamedSNOMEDCode } from '@/utils/fhirConversionUtils';
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: All commented code has been removed.

File: src/components/SearchForm/SearchFormOptions.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: renamed to CancerTypeDetailType

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 29]
```diff
@@ -0,0 +1,145 @@
...
+export type cancerTypeDetails = {
+  category: string;
+  cancerCodes: NamedSNOMEDCode[];
+  cancerSubtype: NamedSNOMEDCode[];
+  biomarkers: NamedSNOMEDCode[];
+  surgeryCodes: NamedSNOMEDCode[];
+  medications: NamedSNOMEDCode[];
+  stages: NamedSNOMEDCode[];
+  radiationCodes: NamedSNOMEDCode[];
+};
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: Modified code as follows
`export type CancerTypeDetailType = {
  category: string;
} & Record<
  'cancerCodes' | 'cancerSubtype' | 'biomarkers' | 'surgeryCodes' | 'medications' | 'stages' | 'radiationCodes',
  CodedValueType[]
`>;``

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 30]
```diff
@@ -0,0 +1,145 @@
...
+export type cancerTypeObj = {
+  lung: cancerTypeDetails;
+  breast: cancerTypeDetails;
+  prostate: cancerTypeDetails;
+  brain: cancerTypeDetails;
+  colon: cancerTypeDetails;
+  mm: cancerTypeDetails;
+};
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: Modified code as follows: 

`export type CancerTypeDetailType = {
  category: string;
} & Record<
  'cancerCodes' | 'cancerSubtype' | 'biomarkers' | 'surgeryCodes' | 'medications' | 'stages' | 'radiationCodes',
  CodedValueType[]
>;`

File: src/components/SearchForm/SearchFormOptions.tsx [hunk 31]
```diff
@@ -0,0 +1,145 @@
...
+export const cancerTypeDetails: cancerTypeObj = tmp;
+
+const ctArray: cancerTypeObj[] = [];
+ctArray.push(tmp);
+/*
+ctArray.push(cancerTypeDetails.prostate.cancerCodes);
+ctArray.push(cancerTypeDetails.colon.cancerCodes);
+ctArray.push(cancerTypeDetails.brain.cancerCodes);
+ctArray.push(cancerTypeDetails.lung.cancerCodes);*/
+export const cancerTypeOptions = ctArray;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/components/SearchForm/SearchFormOptions.tsx]: Previously modified -

File: src/pages/api/clinical-trial-search.ts [hunk 32]
```diff
@@ -123,16 +123,17 @@ function buildBundle(searchParams: SearchParameters): Bundle {
...
     const id = 'mcode-cancer-stage-group';
     const profileValue = fhirConstants.MCODE_CANCER_STAGE_GROUP;
     const codingSystem = 'http://loinc.org';
     const codingSystemCode = '21914-7';
-    convertStringtoResource({
+    console.log('processing stage value=' + JSON.stringify(stageParm));
+    convertNamedSNOMEDCodetoResource({
       bundle: patientBundle,
-      valueString: stageParm,
+      codedValue: stageParm,
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/pages/api/clinical-trial-search.ts]: This function was recoded.  The reference to stageParm was removed.

File: src/pages/api/clinical-trial-search.ts [hunk 33]
```diff
@@ -149,7 +150,7 @@ function buildBundle(searchParams: SearchParameters): Bundle {
     const codingSystemCode: string = null;
     convertStringtoResource({
       bundle: patientBundle,
-      valueString: stageParm,
+      valueString: metastasisParm,
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/pages/api/clinical-trial-search.ts]: Suggested changed made.

File: src/pages/api/clinical-trial-search.ts [hunk 34]
```diff
@@ -194,30 +199,35 @@ function buildBundle(searchParams: SearchParameters): Bundle {
...
-      valueString: radiationParm,
-      id,
-      profile_value: profileValue,
-      codingSystem,
-      codingSystemCode,
-    });
+    for (let i = 0; i < radiationParm.length; i++) {
+      convertNamedSNOMEDCodetoResource({
+        bundle: patientBundle,
+        codedValue: radiationParm[i],
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/pages/api/clinical-trial-search.ts]: this was previously recoded.  radiationParm replaced by searchOptionValue

File: src/queries/fetchCancerSubtypeCodesQuery.ts [hunk 35]
```diff
@@ -1,6 +1,3 @@
-import type { NamedSNOMEDCode } from '@/utils/fhirConversionUtils';
-import mockCancerCodes from './mockData/cancerCodes.json';
+//const fetchCancerSubtypeCodesQuery = async (): Promise<NamedSNOMEDCode[]> => mockCancerCodes.subtypes;
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/queries/fetchCancerSubtypeCodesQuery.ts]: This functionality was replaced by code in the SearchFormOptions.tsx file.

File: src/queries/mockData/breastcancerCodes.json [hunk 36]
```diff
@@ -0,0 +1,177 @@
+{
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/queries/mockData/breastcancerCodes.json:1]: This file is no longer used.  The currently used file is breastCancerTypeCodes.json

File: src/pages/api/clinical-trial-search.ts [hunk 37]
```diff
@@ -71,7 +71,7 @@ function buildBundle(searchParams: SearchParameters): Bundle {
       patient.birthDate = (new Date().getUTCFullYear() - age).toString();
     }
   }
-
+  console.log('***********Got Here1*******');
```
[the remark this answers is not in the document]
Comment (napolr1) [reply src/pages/api/clinical-trial-search.ts]: All console.log calls that I used for debugging have been removed.

**Comment by zlister** [comment]
Closing PR and moving to #43

**Pull request closed without merging**